main.go

```diff
@@ -79,13 +80,12 @@ func modifyPath(arg0path string) error {
 
 		dir, _ = filepath.Abs(dir)
 		path := filepath.Join(dir, pkgConfigExecName)
-		if arg0path == path {
+		if arg0path != path {
 			// Modify the list to exclude the current element and break out of the loop.
-			list = list[i+1:]
-			break
+			filteredList = append(filteredList, dir)
```

Comment:
Can you modify the code to do an in-place remove instead? You can do an in-place move by iterating through the array moving forwards (it currently goes backwards) and the following:

```
copy(list[i:], list[i+1:])
list = list[:len(list)-1]
```

There's no reason for us to move backwards through the path if we are filtering the entire list and we aren't truncating the list.